csrc/id_model/id_model.cpp

```diff
@@ -166,7 +166,7 @@ ValGraph& IdModel::idGraph(IdMappingMode mode) {
 void IdModel::buildIterDomainDefinitionsAndUses() {
   for (const auto tv : tvs_) {
     VectorOfUniqueEntries<IterDomain*> root_domain_ids{
-        tv->getRootDomain().begin(), tv->getRootDomain().end()};
+        tv->getMaybeRootDomain().begin(), tv->getMaybeRootDomain().end()};
```

Comment:
According to how we use IterDomains (when we create a new TV, we always create new IDs, instead of reusing existing IDs), yes, we can. But if we do not want to assume this property, then probably we should still keep this, as we do not know if an ID in the root domain comes from an ID expr somewhere.

Reply:
Ah, I'm sorry, I just thought it's not used at all, but it is indeed used. Sorry for the noise.